mw_virtual_endpoint.go

```diff
@@ -245,7 +251,12 @@ func (d *VirtualEndpoint) ProcessRequest(w http.ResponseWriter, r *http.Request,
 }
 
 func (d *VirtualEndpoint) HandleResponse(rw http.ResponseWriter, res *http.Response, ses *SessionState) error {
+	// Externalising this from the MW so we can re-use it elsewhere
+	return handleForcedResponse(rw, res, ses)
+
+}
 
+func handleForcedResponse(rw http.ResponseWriter, res *http.Response, ses *SessionState) error {
```

Comment:
This func never errors. Either remove the return value, or actually pass on errors like the one from `io.Copy`.